src/dxvk/dxvk_presenter.cpp

```diff
@@ -422,6 +452,69 @@ namespace dxvk {
       m_vkd->vkSetHdrMetadataEXT(m_vkd->device(), 1, &m_swapchain, &hdrMetadata);
   }
 
+  VkResult Presenter::setLatencySleepMode(bool lowLatencyMode, bool lowLatencyBoost, uint32_t minimumIntervalUs) {
+    VkLatencySleepModeInfoNV sleepModeInfo = { VK_STRUCTURE_TYPE_LATENCY_SLEEP_MODE_INFO_NV };
+    sleepModeInfo.lowLatencyMode  = lowLatencyMode;
+    sleepModeInfo.lowLatencyBoost = lowLatencyBoost;
+    sleepModeInfo.minimumIntervalUs = minimumIntervalUs;
+
+    std::lock_guard<dxvk::mutex> lock(m_lowLatencyMutex);
+    VkResult status = m_vkd->vkSetLatencySleepModeNV(m_vkd->device(), m_swapchain, &sleepModeInfo);
+
+    m_lowLatencyEnabled = lowLatencyMode;
+    m_lowLatencyBoost = lowLatencyBoost;
+    m_minimumIntervalUs = minimumIntervalUs;
+
+    if (!lowLatencyMode)
+      m_device->resetLatencyMarkers();
+
+    return status;
+  }
+
+  VkResult Presenter::latencySleep() {
+    VkSemaphore sem = m_lowLatencyFence.fence->handle();
+    uint64_t waitValue = m_lowLatencyFence.value + 1;
+    m_lowLatencyFence.value++;
+
+    VkLatencySleepInfoNV sleepInfo = { VK_STRUCTURE_TYPE_LATENCY_SLEEP_INFO_NV };
+    sleepInfo.signalSemaphore = sem;
+    sleepInfo.value = waitValue;
+
+    {
+      std::lock_guard<dxvk::mutex> lock(m_lowLatencyMutex);
+      m_vkd->vkLatencySleepNV(m_vkd->device(), m_swapchain, &sleepInfo);
+    }
+
+    m_lowLatencyFence.fence->wait(waitValue);
+
+    return VK_SUCCESS;
+  }
+
+  void Presenter::setLatencyMarker(VkLatencyMarkerNV marker, uint64_t presentId) {
+    VkSetLatencyMarkerInfoNV markerInfo = { VK_STRUCTURE_TYPE_SET_LATENCY_MARKER_INFO_NV };
+    markerInfo.presentID = presentId;
+    markerInfo.marker = marker;
+
+    std::lock_guard<dxvk::mutex> lock(m_lowLatencyMutex);
+    m_vkd->vkSetLatencyMarkerNV(m_vkd->device(), m_swapchain, &markerInfo);
+  }
+
+  VkResult Presenter::getLatencyTimings(std::vector<VkLatencyTimingsFrameReportNV>& frameReports) {
+    VkGetLatencyMarkerInfoNV markerInfo = { VK_STRUCTURE_TYPE_GET_LATENCY_MARKER_INFO_NV };
+    uint32_t timingCount = 0;
+
+    std::lock_guard<dxvk::mutex> lock(m_lowLatencyMutex);
+    m_vkd->vkGetLatencyTimingsNV(m_vkd->device(), m_swapchain, &timingCount, &markerInfo);
+
+    if (timingCount != 0) {
+      frameReports.resize(timingCount, { VK_STRUCTURE_TYPE_GET_LATENCY_MARKER_INFO_NV });
+      markerInfo.pTimings = frameReports.data();
+
+      m_vkd->vkGetLatencyTimingsNV(m_vkd->device(), m_swapchain, &timingCount, &markerInfo);
```

Comment:
I think `VkStructureType` passed to `resize` is wrong here (`VK_STRUCTURE_TYPE_GET_LATENCY_MARKER_INFO_NV` → `VK_STRUCTURE_TYPE_LATENCY_TIMINGS_FRAME_REPORT_NV`)? So together with updated Vulkan headers for spec 2 version of NV_ll2 this should look more like:

```suggestion

    std::lock_guard<dxvk::mutex> lock(m_lowLatencyMutex);
    m_vkd->vkGetLatencyTimingsNV(m_vkd->device(), m_swapchain, &markerInfo);

    if (markerInfo.timingCount != 0) {
      frameReports.resize(markerInfo.timingCount, { VK_STRUCTURE_TYPE_LATENCY_TIMINGS_FRAME_REPORT_NV });
      markerInfo.pTimings = frameReports.data();

      m_vkd->vkGetLatencyTimingsNV(m_vkd->device(), m_swapchain, &markerInfo);
```

Reply:
Good catch, this should be fixed now.